spock-core/src/main/java/org/spockframework/runtime/IRunListener.java

```diff
@@ -31,6 +31,11 @@
   void beforeSpec(SpecInfo spec);
 
   /**
+   * Called when a block is reached.
```

Comment:
Don't have a use case for afterBlock. My current use case for this is to have dinamic reports by resolving the block expression before sending it to the reports listener